### PR TITLE
chore: upgrade smithy to 1.13

### PR DESCRIPTION
**Pull request by aajtodd**
## Issue \#

N/A
## Description of changes

* Upgrade to latest version of smithy
* Disable new protocol tests that are failing (issues created and added to M5 milestone)
* Fix generation of sdk-version.txt to use gradle.properties as an input (should rerun if changed now)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -7,9 +7,5 @@
 <!--- Why is this change required? What problem does it solve? -->
 
 
-## Scope
-<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
-- [ ] Bug fix (non-breaking change which fixes an issue)
-- [ ] New feature (non-breaking change which adds functionality)
```
Comment (ianbotsf) [inline .github/pull_request_template.md:13]: **Correctness**: Unrelated to upgrading Smithy. This should be moved to a separate commit.

**Review by ianbotsf — APPROVED** [review]
Minor note about a file that doesn't belong in this commit. Otherwise, looks good.

**Review by kggilmer — APPROVED** [review]
